compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPULowerToUKernels.cpp

```diff
@@ -189,12 +358,48 @@ struct UKernelMmt4DPattern : OpRewritePattern<linalg::Mmt4DOp> {
     return success();
   }
 };
+
+/// Pattern to lower tensor.pack to ukernel.generic
+struct UKernelPackPattern : OpRewritePattern<tensor::PackOp> {
```

Comment:
Create a single template class for all these patterns. They are literally the same code.